src/grpc/rpc_server.cpp

```diff
@@ -155,11 +155,23 @@ bool GrpcServer::run_generic_handler_cb(const std::string& rpc_name, boost::intr
             // respond immediately
             return true;
         }
-        cb = it->second;
+        cb = it->second.first;
     }
     return cb(rpc_data);
 }
 
+void GrpcServer::run_generic_completion_cb(const std::string& rpc_name,
+                                           boost::intrusive_ptr< GenericRpcData >& rpc_data) {
+    generic_rpc_completed_cb_t cb;
+    {
+        std::shared_lock< std::shared_mutex > lock(m_generic_rpc_registry_mtx);
+        auto it = m_generic_rpc_registry.find(rpc_name);
+        if (it == m_generic_rpc_registry.end() || !(it->second.second)) { return; }
```

Comment:
it != m_generic_rpc_registry.end() can be put in an assert.

Reply:
if rpc_name is not found in map, can we avoid acquire lock again, since we already have knowledge in handler_cb context this is not found in map.